packages/utils/src/logger.ts

```diff
@@ -43,42 +22,90 @@ export interface BaseLogObject {
     req?: express.Request;
 }
 
+export type LogLevel = "debug" | "info" | "warn" | "error" | "silent";
+
 export class Logger {
+    public static readonly LEVEL_DEBUG = pino.levels.values.debug;
+    public static readonly LEVEL_INFO = pino.levels.values.info;
+    public static readonly LEVEL_WARN = pino.levels.values.warn;
+    public static readonly LEVEL_ERROR = pino.levels.values.error;
+    public static readonly LEVEL_SILENT = pino.levels.values.silent;
+
+    /**
+     * Returns the corresponding numerical value of the log level corresponding to `levelName`.
+     * @param levelName
+     * @throws ArgumentError if an invalid log level name is provided.
+     */
+    public static getLevel(levelName: string) {
+        switch (levelName) {
+            case "debug":
+                return Logger.LEVEL_DEBUG;
+            case "info":
+                return Logger.LEVEL_INFO;
+            case "warn":
+                return Logger.LEVEL_WARN;
+            case "error":
+                return Logger.LEVEL_ERROR;
+            case "silent":
+                return Logger.LEVEL_SILENT;
+            default:
+                throw new ArgumentError(`"${levelName}" is not a valid log level name.`);
+        }
+    }
+
+    private readonly children: Logger[] = [];
+
     private constructor(private readonly pino: pino.Logger) {}
 
-    public static getLogger() {
+    public static getLogger(name = "not-set", level: LogLevel = enabled ? "debug" : "silent") {
```

Comment:
How will this work with the Config<T> params - since we want to be able to change the log level. Maybe we need to move the Config<T> object into the pisa repo?

Reply:
I'm not sure moving the Config here solves the issue: to change the config level, you need to explicitly set pino.level; instead, the Config level silently changes the value, so it works for settings where we read that value every once in a while.

If we want to use the Config, I guess the options are:
- have a timer to periodically check the logLevel in the Config, and update pino if needed
- do the check every time a function in our Logger is called, before forwarding to pino.

Both seem bad hacks, though. I'd rather put this on a higher level (that is, the place that reads the service config and creates the logger object, sets the level explicitly every time the config is reloaded.

Reply:
kk, sounds good